protocol/transport/transport.go

```diff
@@ -717,3 +717,41 @@ func (t *Transport) RemovePubsubTopicKey(topic string) error {
 	}
 	return nil
 }
+
+func (t *Transport) CheckForMissingMessages(storenode []byte, filters []*Filter) {
+	if t.waku.Version() != 2 {
+		return
+	}
+
+	topicMap := make(map[string]map[string]struct{})
+	for _, f := range filters {
+		_, ok := topicMap[f.PubsubTopic]
+		if !ok {
+			topicMap[f.PubsubTopic] = make(map[string]struct{})
+		}
+
+		topicMap[f.PubsubTopic][f.ContentTopic.String()] = struct{}{}
+	}
+
+	peer, err := peer.Decode(string(storenode))
+	if err != nil {
+		t.logger.Error("could not decode peerID", zap.Error(err))
+		return
+	}
+
+	for pubsubTopic, contentTopics := range topicMap {
+		ctList := []string{}
+		for ct := range contentTopics {
+			ctList = append(ctList, ct)
+		}
+		err = t.waku.CheckForMissingMessages(peer, pubsubTopic, ctList)
+		if err != nil {
+			t.logger.Error("could not check for missing messages",
+				zap.Error(err),
+				zap.Stringer("peerID", peer),
+				zap.String("pubsubTopic", pubsubTopic),
+				zap.Strings("contentTopics", ctList))
```

Comment:
I would keep track of the last successful `checkformissingmessages` query and do a time range query from said time to now to not overwhelm the store servers.

Reply:
Done in https://github.com/status-im/status-go/blob/d97e4508abf8fad886c0e226e8e12a2c62bbc2ac/wakuv2/missing_messages.go#L126 
This value is updated only if the query was succesful

Reply:
awesome (not sure if it was already there, getting used to Golang :) )